src/tilesource.js

```diff
@@ -275,28 +276,39 @@ $.TileSource.prototype = {
         return this._tileHeight;
     },
 
+    /**
+     * @function
+     * @param {Number} level
+     * @return {OpenSeadragon.TileSource} Chainable.
+     */
+    setMaxLevel: function( level ) {
+        this.maxLevel = level;
+        this._memoizeLevelScale( level );
+        return this;
+    },
+
     /**
      * @function
      * @param {Number} level
      */
     getLevelScale: function( level ) {
+        this._memoizeLevelScale( this.maxLevel );
```

Comment:
It might be worth adding a little comment here (not in the doc comment but just here in the code) about how this works (the fact that this function is only called the first time and then the memoize makes a new version of it), since it can be a little confusing otherwise.